### DISCUSSION TITLE
Saving the model

### DISCUSSION THREAD

**Discussion by iRanadheer** [neonbjb/tortoise-tts#155] [General]
Hey, 

your API looks really great! I'm able to train the model with a custom voice and generate the output in less than few minutes. 

I have one question though, is it possible to save the model that was trained on the custom voice? We dont want to train the model everytime when we change the text. 

Thank you

**Comment by neonbjb** [comment]
Hey there,
I'm not quite sure what you mean by "train the model". Do you mean "provide a new voice"? There is no training involved with giving the model a new voice, and the overhead is minimal.

**Reply by FurkanGozukara** [reply]
@neonbjb thank you so much for this amazing open source. How much new voice we should provide. I have excellent recordings already long enough on my youtube channel. I Want replace my old video sound entirely

**Comment by iRanadheer** [comment]
Hey,

yes, giving the new voice to the model. I have few thousands of audio clips and I see we are converting all the audio files into tensors. Is there any reason we can't save the model (or tensors) and just run the tts_with_preset function?

**Reply by neonbjb** [reply]
The per-voice preprocessing time is negligible compared to some of the other latencies in this stack. Tortoise isn't really optimized for production use, nor do I intend to do any work in that direction. I would be glad to review PRs if you find any optimizations you want to share with the community.

**Reply by WentingMiao** [reply]
Hi, I am also interested in it. Could you share any optimization if you find one?

**Reply by dekker33** [reply]
Can someone please tell me what I am doing wrong....
I have a folder called "train_trump" in the "voices" directory, with 3 ~10 second long recordings of trump talking, all floating point .wavs with a 22050 sample rate.

Anaconda Prompt:

C:\Users\me\anaconda3\tortoise-tts>python tortoise/do_tts.py --text "the virus is from china" --voice train_trump --preset fast
C:\Users\me\anaconda3\tortoise-tts\tortoise\utils\audio.py:17: WavFileWarning: Chunk (non-data) not understood, skipping it.
sampling_rate, data = read(full_path)
Traceback (most recent call last):
File "tortoise/do_tts.py", line 35, in
voice_samples, conditioning_latents = load_voices(voice_sel)
File "C:\Users\me\anaconda3\tortoise-tts\tortoise\utils\audio.py", line 124, in load_voices
clip, latent = load_voice(voice, extra_voice_dirs)
File "C:\Users\me\anaconda3\tortoise-tts\tortoise\utils\audio.py", line 111, in load_voice
c = load_audio(cond_path, 22050)
File "C:\Users\me\anaconda3\tortoise-tts\tortoise\utils\audio.py", line 31, in load_audio
audio, lsr = load_wav_to_torch(audiopath)
File "C:\Users\me\anaconda3\tortoise-tts\tortoise\utils\audio.py", line 25, in load_wav_to_torch
raise NotImplemented(f"Provided data dtype not supported: {data.dtype}")
TypeError: 'NotImplementedType' object is not callable
